solidity/random-beacon/test/RandomBeacon.Pool.test.ts

```diff
@@ -21,67 +25,94 @@ describe("RandomBeacon - Pool", () => {
 
   beforeEach("load test fixture", async () => {
     const contracts = await waffle.loadFixture(fixture)
+
+    randomBeacon = contracts.randomBeacon as RandomBeaconStub
     sortitionPool = contracts.sortitionPool as SortitionPool
     stakingStub = contracts.stakingStub as StakingStub
-    randomBeacon = contracts.randomBeacon as RandomBeacon
   })
 
-  describe("registerMemberCandidate", () => {
-    const minimumStake = 2000
+  describe("registerOperator", () => {
     beforeEach(async () => {
-      await stakingStub.setStake(operator.getAddress(), minimumStake)
+      await stakingStub.setStake(operator.address, constants.minimumStake)
     })
 
     context("when the operator is not registered yet", () => {
       beforeEach(async () => {
-        await randomBeacon.connect(operator).registerMemberCandidate()
+        await randomBeacon.connect(operator).registerOperator()
+      })
+
+      it("should deposit gas", async () => {
+        expect(await randomBeacon.hasGasDeposit(operator.address)).to.be.true
       })
 
       it("should register the operator", async () => {
-        await expect(
-          await sortitionPool.isOperatorInPool(await operator.getAddress())
-        ).to.be.true
+        expect(await sortitionPool.isOperatorInPool(operator.address)).to.be
+          .true
       })
     })
 
     context("when the operator is already registered", () => {
       beforeEach(async () => {
-        await randomBeacon.connect(operator).registerMemberCandidate()
+        await randomBeacon.connect(operator).registerOperator()
       })
 
       it("should revert", async () => {
         await expect(
-          randomBeacon.connect(operator).registerMemberCandidate()
+          randomBeacon.connect(operator).registerOperator()
         ).to.be.revertedWith("Operator is already registered")
       })
     })
   })
 
+  describe("updateOperatorStatus", () => {
+    let operatorID: number
+
+    beforeEach(async () => {
+      // Operator is registered and gas deposit is made.
+      await stakingStub.setStake(operator.address, constants.minimumStake)
+      await randomBeacon.connect(operator).registerOperator()
+      operatorID = await sortitionPool.getOperatorID(operator.address)
+
+      // Simulate the operator became ineligible.
+      await stakingStub.setStake(operator.address, 0)
+
+      await randomBeacon.connect(operator).updateOperatorStatus()
+    })
+
+    it("should remove operator from the pool", async () => {
+      expect(await sortitionPool.isOperatorInPool(operator.address)).to.be.false
+    })
+
+    it("should release the gas deposit if operator was removed from pool during the update", async () => {
+      expect(await randomBeacon.hasGasDeposit(operator.address)).to.be.false
+    })
```

Comment:
We are missing some `context()` here, `updateOperatorStatus()` does not remove anyone from the pool just because it's called. In this case, the eligible stake went down to 0.

Reply:
Done in https://github.com/keep-network/keep-core/pull/2707/commits/c97dba585f021ce3e8ec7fb37e2665a4f7577541 as part of #2707